rust/lance-encoding/src/encodings/physical/fixed_size_binary.rs

```diff
@@ -84,7 +84,7 @@ impl PrimitivePageDecoder for FixedSizeBinaryDecoder {
         let num_bytes = num_rows * self.byte_width;
         let bytes = self.bytes_decoder.decode(rows_to_skip, num_bytes)?;
         let bytes = bytes.as_fixed_width().unwrap();
-        debug_assert_eq!(bytes.bits_per_value, 8);
+        debug_assert_eq!(bytes.bits_per_value, self.byte_width * 8);
```

Comment:
This decoder will be used if Lance format 2.1 is enabled and this assertion will fail for some input data.